ERCS/erc-7955.md

```diff
@@ -27,27 +27,27 @@ There are currently three main approaches to this problem:
-- Reverts due to alternative gas schedules make the CREATE2 factory no longer deployable.
+- Reverts, due to alternative gas schedules for example, make the CREATE2 factory no longer deployable.
 
 #### 2. Secret Private Key
 
-Keep a carefully guarded secret key and use it to sign transactions to deploy CREATE2 factory contracts. The resulting contract will have a deterministic address and code on all chains where the first transaction of the deployer account is a CREATE2 factory deployment, which can be verified post-deployment to ensure trustlessness. Additionally, this method does not have the same gas sensitivity downsides as Nick's method, as the private key can sign a creation transaction with appropriate gas parameters at the time of execution. This is the strategy used by the _Safe Singleton Factory_ (`safe-global/safe-singleton-factory`) and _CreateX_ (`pcaversaccio/createx`).
+Keep a carefully guarded secret key and use it to sign transactions to deploy CREATE2 factory contracts. The resulting contract will have a deterministic address and code on all chains where the transaction at a given nonce of the deployer account is a CREATE2 factory deployment, which can be verified post-deployment to ensure trustlessness. Additionally, this method does not have the same gas sensitivity downsides as Nick's method, as the private key can sign a creation transaction with appropriate gas parameters at the time of execution. This is the strategy used by the _Safe Singleton Factory_ and _CreateX_ (deployed to `0x914d7Fec6aaC8cd542e72Bca78B30650d45643d7` and `0xba5Ed099633D3B313e4D5F7bdc1305d3c28ba5Ed` respectively, including on Ethereum Mainnet).
 
 **Downsides**:
 
 - It is permissioned: the party that holds the secret key has the ultimate say on which chains will get the CREATE2 factory deployments.
 - This requires carefully guarding a secret key; if it is exposed or lost, deployments are no longer guaranteed on new chains.
-- If the first transaction is not a successful CREATE2 factory deployment, then it is no longer possible to have a CREATE2 factory at the common address; this can happen by human error, for example.
+- If the transaction at the given nonce is not a successful CREATE2 factory deployment, then it is no longer possible to have a CREATE2 factory at the canonical address; this can happen by human error, for example.
 
 #### 3. Preinstalls
 
-Have popular CREATE2 deployment factories deployed on new chains by default. This is, for example, what OP Stack does as part of their preinstalls, including the CREATE2 factory contracts mentioned above. This ensures that the CREATE2 factory contracts have known addresses and codes.
+Have popular CREATE2 deployment factories deployed on new chains by default. This is, for example, what OP Stack and ZKsync do as part of their preinstalls, including the CREATE2 factory contracts mentioned above. This ensures that the CREATE2 factory contracts have known addresses and codes.
```

Comment:
I think Arbitrum and Polygon do this too, but best check with @Fbartoli

Reply:
> Arbitrum

Doing a quick search in the arbitrum repositories, it does not _appear_ to be a preinstall (<https://github.com/search?q=org%3AOffchainLabs+0x4e59b44847b379578588920ca78fbf26c0b4956c&type=code>)

> Polygon

And same for the Polygon repos (<https://github.com/search?q=org%3Amaticnetwork%200x4e59b44847b379578588920ca78fbf26c0b4956c&type=code>)